calendar/src/main/java/com/toggl/calendar/calendarday/domain/CalendarDayReducer.kt

```diff
@@ -3,21 +3,27 @@ package com.toggl.calendar.calendarday.domain
 import com.toggl.architecture.core.Effect
 import com.toggl.architecture.core.MutableValue
 import com.toggl.architecture.core.Reducer
+import com.toggl.architecture.extensions.effect
 import com.toggl.calendar.common.domain.toSelectedCalendarItem
 import com.toggl.common.feature.extensions.mutateWithoutEffects
+import com.toggl.environment.services.calendar.CalendarService
 import javax.inject.Inject
 import javax.inject.Singleton
 
 @Singleton
-class CalendarDayReducer @Inject constructor() : Reducer<CalendarDayState, CalendarDayAction> {
+class CalendarDayReducer @Inject constructor(private val calendarService: CalendarService) : Reducer<CalendarDayState, CalendarDayAction> {
 
     override fun reduce(
         state: MutableValue<CalendarDayState>,
         action: CalendarDayAction
     ): List<Effect<CalendarDayAction>> =
         when (action) {
-                is CalendarDayAction.ItemTapped -> state.mutateWithoutEffects {
+            is CalendarDayAction.ItemTapped -> state.mutateWithoutEffects {
                 copy(selectedItem = action.calendarItem.toSelectedCalendarItem())
             }
+            CalendarDayAction.CalendarViewAppeared -> effect(FetchCalendarEventsEffect(calendarService, state().date, state().date, state().calendars))
```

Comment:
What do you think about? Btw, this line will break if you auto format what you've written.
```suggestion
            CalendarDayAction.CalendarViewAppeared -> effect(state().run {
                FetchCalendarEventsEffect(calendarService, date, date, calendars)
            })
```